packages/hop-node/src/gasboost/GasBoostSigner.ts

```diff
@@ -46,72 +47,86 @@ class GasBoostSigner extends Wallet {
       `GasBoostSigner, label: ${prefix}, host: ${hostname}`
     )
     this.setOptions(options)
-    this.restore()
+    this.init()
+      .catch((err: Error) => this.logger.error('init error:', err))
+      .finally(async () => {
+        const nonce = await this.getDbNonce()
+        this.logger.debug('ready')
+        this.logger.debug(`current nonce: ${nonce}`)
+        this.ready = true
+      })
   }
 
-  setStore (store: Store) {
-    this.store = store
+  private async init () {
+    await this.setLatestNonce()
+  }
+
+  protected async tilReady (): Promise<boolean> {
+    if (this.ready) {
+      return true
+    }
+
+    await wait(100)
+    return await this.tilReady()
+  }
+
+  private async setLatestNonce () {
```

Comment:
IMO the nonce from the initialization of the bonder should only come from on-chain and should not care about the DB. I cannot think of a case* where the DB would be correct but on-chain would be incorrect.

Reply:
it will cause issues if the stored nonces are being overridden by separate processes which defeats the purpose

a scenario,
1. bonder is running as usual
3. a bonder command (like a stake command) is ran to send tx
4. right before the bonder command sends tx, the running bonder sends tx with nonce #10 (ie a usual bond withdawal tx)
5. the bonder command sends tx with nonce #10 as well because it fetched on-chain nonce within seconds of last tx send instead of using db nonce which would of been nonce #11. The bonder command tx used a slightly higher gas price so the running bonder tx fails and causes 1hr delay 

related: https://github.com/hop-protocol/hop/pull/178#issuecomment-929880265

Reply:
I see. I think there is another scenario where a tx gets re-orged out (although I believe the bug has been fixed) and the DB is out of sync. In that case, I think there is no way for the bonder to recover, since a new instantiation would use the DB state. Right now, we `sendToSelf`, which uses on-chain state and fixes the issue, but that would not be possible with this update.

If what I said is true, then I'm pretty torn. I'd lean towards still not doing `max()` and only using on-chain for init _only_ because it can get the bonder into an unrecoverable state (either the re-org mentioned above or a potential yet-unknown bug with the bonder software or RPC provider). I do not believe we should not allow the bonder to get in an unrecoverable state, if possible.

I _think_ the only downside of not using `max()` is that a user waits 1 hour if the timing is wrong.

LMK what you think, pretty open to discussion here.

Reply:
That makes sense 👍  I'll update it accordingly to set pending nonce on start

```js
  private async setLatestNonce () {
    const onChainNonce = await this.getOnChainNonce()
    await this.setDbNonce(onChainNonce)
}
```